wsr-tic-tac-toe/src/__tests__/game-logic.test.ts

```diff
@@ -0,0 +1,20 @@
+import { generateWinningScenarios } from '../utils/game-logic';
+
+test('negative board sizes should return no winning scenarios', () => {
+    expect(generateWinningScenarios(-10)).toEqual([]);
+});
+
+test('board size: 0 should return no winning scenarios', () => {
+    expect(generateWinningScenarios(0)).toEqual([]);
+});
+
+test('board size: 1 should return one winning scenario', ()=> {
+    expect(generateWinningScenarios(1)).toHaveLength(1);
+})
+
+test('board size: n (n>1) should return n*2+2 winning scenarios. (n=[2,47])', ()=>{
```

Comment:
add a test to check content of winningScenarios, not only how many were generated.

You don't have to check all, although with something like 2x2 board it may even be fine